InsertionSortProj/InsertionSort/InsertionSort.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace InsertionSortProj
+{
+    public class InsertionSort
+    {
+        private static void Main(string[] args)
+        {
+            Console.WriteLine("Insert array to sort");
+            try
+            {
+                var res = ConsoleArrayRead();
+                Sort(res);
+                Console.WriteLine("There is your sorted array");
+                Console.WriteLine("[{0}]", string.Join(", ", res));
+            }
+            catch (Exception ex)
+            {
+                switch (ex)
+                {
+                    case FormatException e:
+                        Console.WriteLine("Failed! Example of input: -1 0 1 2");
+                        break;
+                    case ArgumentException e:
+                        Console.WriteLine("Expected array of integers");
+                        break;
+                    default:
+                        Console.WriteLine("Unknown exception");
+                        break;
+                }
+            }
+        }
+
+        public static void Sort(int[] arr)
+        {
+            for (int sortedLen = 0; sortedLen < arr.Length; sortedLen++)
+            {
+                for (int k = 0; k < sortedLen; k++)
+                {
+                    int i = sortedLen - k;
+                    if (arr[i] < arr[i - 1])
+                    {
+                        (arr[i], arr[i-1]) = (arr[i-1], arr[i]);
+                    } else
+                    {
+                        break;
+                    }
+                }
+            }
+        }
+      
+        private static int[] ConsoleArrayRead()
+        {
+            var numbers = Console.ReadLine().Split(' ');
```

Comment:
CI огорчает тот факт, что ReadLine() может вернуть null, но тут на это не проверяется. Могло бы помочь так: 
```suggestion
            var numbers = Console.ReadLine()?.Split(' ');
```
Но тогда numbers надо было бы проверить на null. Но это нормально, код должен быть устойчив к ошибкам.